WordPress/src/main/java/org/wordpress/android/ui/history/HistoryDetailContainerFragment.java

```diff
@@ -172,6 +179,37 @@ public void onClick(View view) {
         return rootView;
     }
 
+    private void mapRevisions() {
+        if (getArguments() != null) {
+            mRevision = getArguments().getParcelable(EXTRA_CURRENT_REVISION);
+
+            final long[] previousRevisionsIds = getArguments().getLongArray(EXTRA_PREVIOUS_REVISIONS_IDS);
+            final List<RevisionModel> revisionModels = new ArrayList<>();
+            for (final long revisionId : previousRevisionsIds) {
+                revisionModels.add(mPostStore.getRevisionById(revisionId));
+            }
+            mRevisions = mapRevisionModelsToRevisions(revisionModels);
+        }
+    }
+
+    @Nullable
+    private ArrayList<Revision> mapRevisionModelsToRevisions(@Nullable final List<RevisionModel> revisionModels) {
+        if (revisionModels == null) {
+            return null;
+        }
+        final ArrayList<Revision> revisions = new ArrayList<>();
+        for (int i = 0; i < revisionModels.size(); i++) {
+            final RevisionModel current = revisionModels.get(i);
+            final Revision revision = new Revision(current.getRevisionId(), current.getDiffFromVersion(),
```

Comment:
`Revision` has constructor that takes a `RevisionModel`. Wonder if we can utilize it?